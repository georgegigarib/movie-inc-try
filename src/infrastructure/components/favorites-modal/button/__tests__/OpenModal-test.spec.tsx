import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import OpenModalButton from '@/src/infrastructure/components/favorites-modal/button/OpenModal';
  
describe('OpenModalButton', () => {
  it('matches the snapshot when isFavorite is true', () => {
    const { toJSON } = render(<OpenModalButton isFavorite={true} onPress={jest.fn()} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches the snapshot when isFavorite is false', () => {
    const { toJSON } = render(<OpenModalButton isFavorite={false} onPress={jest.fn()} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the correct icon based on isFavorite prop', () => {
    const { getByTestId, rerender } = render(<OpenModalButton isFavorite={true} onPress={jest.fn()} />);

    const heartIcon = getByTestId('favorite-icon');
    expect(heartIcon.props.children).toBe('heart');
    expect(heartIcon.props.style.fontSize).toBe(15);
    expect(heartIcon.props.style.color).toBe('#fff');

    rerender(<OpenModalButton isFavorite={false} onPress={jest.fn()} />);

    const heartOutlineIcon = getByTestId('favorite-icon');
    expect(heartOutlineIcon.props.children).toBe('heart-o');
    expect(heartOutlineIcon.props.style.fontSize).toBe(15);
    expect(heartOutlineIcon.props.style.color).toBe('#fff');
  });

  it('calls onPress function when button is pressed', () => {
    const mockOnPress = jest.fn();
    const { getByTestId } = render(<OpenModalButton isFavorite={false} onPress={mockOnPress} />);

    const button = getByTestId('open-modal-button');
    fireEvent.press(button);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
