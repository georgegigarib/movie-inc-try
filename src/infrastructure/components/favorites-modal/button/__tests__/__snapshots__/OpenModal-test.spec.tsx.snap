// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenModalButton matches the snapshot when isFavorite is false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "borderRadius": 20,
      "opacity": 1,
      "padding": 10,
      "position": "absolute",
      "right": 10,
      "top": 10,
    }
  }
  testID="open-modal-button"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 15,
      }
    }
    testID="favorite-icon"
  >
    heart-o
  </Text>
</View>
`;

exports[`OpenModalButton matches the snapshot when isFavorite is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "borderRadius": 20,
      "opacity": 1,
      "padding": 10,
      "position": "absolute",
      "right": 10,
      "top": 10,
    }
  }
  testID="open-modal-button"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 15,
      }
    }
    testID="favorite-icon"
  >
    heart
  </Text>
</View>
`;
